CMakeLists.txt

```diff
@@ -480,39 +480,56 @@ find_package(JSONC REQUIRED)
 # Set variable required by libclamav to use libjson-c
 set(HAVE_JSON 1)
 
+set(LLVM_MAX_VER "13.0.1")
+set(LLVM_MIN_VER "8.0.0")
+
 string (TOLOWER ${BYTECODE_RUNTIME} bytecodeRuntime)
 if(${bytecodeRuntime} STREQUAL "llvm")
-    set (LLVM_FIND_VERSION "8.0.0")
-    find_package(LLVM REQUIRED)
+    if(DEFINED LLVM_ROOT_DIR AND DEFINED LLVM_FIND_VERSION)
+        find_package(LLVM EXACT ${LLVM_FIND_VERSION} REQUIRED HINTS ${LLVM_ROOT_DIR})
+    elseif(DEFINED LLVM_ROOT_DIR)
+        find_package(LLVM REQUIRED HINTS ${LLVM_ROOT_DIR})
+    elseif(DEFINED LLVM_FIND_VERSION)
+        find_package(LLVM EXACT ${LLVM_FIND_VERSION} REQUIRED)
+    else()
+        set (LLVM_FIND_VERSION ${LLVM_MIN_VER})
+        find_package(LLVM REQUIRED)
+    endif()
     if(LLVM_FOUND)
         if (LLVM_AVAILABLE_LIBS)
             # Found using LLVMConfig.cmake
-            message("LLVM found using LLVMConfig.cmake")
-            set(LLVM_VERSION ${LLVM_VERSION_MAJOR}${LLVM_VERSION_MINOR})
+            message(STATUS "LLVM found using LLVMConfig.cmake")
+
             set(LLVM_LIBRARIES ${LLVM_AVAILABLE_LIBS})
 
-            if (${LLVM_PACKAGE_VERSION} VERSION_LESS "8.0.0")
-                message(FATAL "LLVM version ${LLVM_PACKAGE_VERSION} is too old")
+            if (${LLVM_PACKAGE_VERSION} VERSION_LESS ${LLVM_MIN_VER})
+                message(FATAL_ERROR "LLVM version ${LLVM_PACKAGE_VERSION} is too cold")
+            elseif (${LLVM_PACKAGE_VERSION} VERSION_GREATER ${LLVM_MAX_VER} )
+                message(FATAL_ERROR "LLVM version ${LLVM_PACKAGE_VERSION} is too hot")
+            else()
+                message(STATUS "LLVM version ${LLVM_PACKAGE_VERSION} is just right")
             endif()
-
         else()
             # Found using FindLLVM.cmake
-            message("LLVM found using FindLLVM.cmake")
-
-            # Set variable required by libclamav to use llvm instead of interpreter
-            set(LLVM_VERSION ${LLVM_VERSION_MAJOR}${LLVM_VERSION_MINOR})
-            message("LLVM_FOUND ${LLVM_FOUND}")
+            message(STATUS "LLVM found using FindLLVM.cmake")
 
             if (${LLVM_VERSION_STRING} VERSION_GREATER_EQUAL "9.0.0")
                 if (${LLVM_VERSION_STRING} VERSION_LESS "10.0.0")
                     set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DNDEBUG")
                 endif()
             endif()
 
-            if (${LLVM_VERSION_STRING} VERSION_LESS "8.0.0")
-                message(FATAL "LLVM version ${LLVM_VERSION_STRING} is too old")
+            if (${LLVM_VERSION_STRING} VERSION_LESS ${LLVM_MIN_VER})
+                message(FATAL_ERROR "LLVM version ${LLVM_VERSION_STRING} is too cold")
+            elseif (${LLVM_VERSION_STRING} VERSION_GREATER ${LLVM_MAX_VER} )
```

Comment:
Instead of checking if `VERSION_GREATER 13.0.1`, let's check if `VERSION_GREATER_EQUAL 14.0.0`.  Presumably we'll suppost 13.0.2 if there is a security patch version, and we'll want any users on an older release to adopt it.